docs/intro/overrides.rst

```diff
@@ -0,0 +1,387 @@
+
+.. code-block::
+
+    Use this                                              instead of                                  for the URL patterns                    except for the patterns      with priority  meta
+    ----------------------------------------------------  ------------------------------------------  --------------------------------------  -------------------------  ---------------  ------
+    my_project.page_objects.ExampleProductPage            my_project.page_objects.GenericProductPage  ['example.com']                         []                                     500  {}
+    my_project.page_objects.AnotherExampleProductPage     my_project.page_objects.GenericProductPage  ['anotherexample.com']                  ['/digital-goods/']                    500  {}
+    my_project.page_objects.DualExampleProductPage        my_project.page_objects.GenericProductPage  ['dualexample.com', 'dualexample.net']  []                                     500  {}
+
+Organizing Page Object Overrides
+--------------------------------
+
+After tackling the two (2) different approaches from the previous chapters on how
+to declare overrides, we can now explore how to organize them in our projects.
+Although it's mostly up to the developer which override declaration method to
+use. Yet, we'll present some approaches depending on the situation.
+
+To put this thought into action, let's suppose we are tasked to create a Page
+Object Project with overrides for eCommerce websites.
+
+Package-based Approach
+~~~~~~~~~~~~~~~~~~~~~~
+
+Using the **package-based** approach, we might organize them into something like:
+
+.. code-block::
+
+    my_page_obj_project
+    ├── cool_gadget_site
+    |   ├── us
+    |   |   ├── __init__.py
+    |   |   ├── products.py
+    |   |   └── product_listings.py
+    |   ├── fr
+    |   |   ├── __init__.py
+    |   |   ├── products.py
+    |   |   └── product_listings.py
+    |   └── __init__.py
+    └── furniture_shop
+        ├── __init__.py
+        ├── products.py
+        └── product_listings.py
+
+Assuming that we've declared the Page Objects in each of the modules to use the
+``default_registry`` like:
+
+.. code-block:: python
+
+    # my_page_obj_project/cool_gadget_site/us/products.py
+
+    from web_poet import handle_urls  # remember that this uses the default_registry
+    from web_poet.pages import ItemWebPage
+
+    @handle_urls("coolgadgetsite.com", overrides=GenericProductPage)
+    class CoolGadgetUsSiteProductPage(ItemWebPage):
+        def to_item(self):
+            ... # parsers here
+
+Then we could easily retrieve all Page Objects per subpackage or module like this:
+
+.. code-block:: python
+
+    from web_poet import default_registry
+
+    # We can do it per website.
+    rules = default_registry.get_overrides_from_module("my_page_obj_project.cool_gadget_site")
+    rules = default_registry.get_overrides_from_module("my_page_obj_project.furniture_site")
```

Comment:
This example suggests that the function should be named `default_registry.get_overrides_from_package`

Reply:
I was thinking about that too.

I believe this was originally named like this to follow `importlib`'s naming convention where it uses the **module** terminology which includes **packages** and **subpackages**. Renaming this into `get_overrides_from_package` also present some slight ambiguity since it also supports **modules**.

I think we can rename it to simply `get_overrides_from` instead. This new name also opens up other possibilities of supporting paths like `my_proj/path/something` if we detect a `/` char. I'm not sure if users would like to have this but it's now an option in the future.

Update this on: https://github.com/scrapinghub/web-poet/pull/16/commits/c000cbc9d36f1c3022c22d27d67d83e4d86b41c7 Let me know if you have some concerns on this.

Reply:
Resolving this thread as it's outdated. The latest commits has a much simplified API and now simply calls it as `get_overrides()`. A param named `filter` is now available to easily filter out packages/modules. Reference: https://github.com/scrapinghub/web-poet/pull/16/files#diff-409cb5fe79e7d7d5ede41109573e31bd847a147a966f477de80c99279bcdf479R249-R251